content/interoperability.tex

```diff
@@ -0,0 +1,191 @@
+from that returned in an \openshmem-only program. For instance, the former
+initialization call in a hybrid program may initialize a resource with the
+user-requested thread level, but the supported level cannot be updated by the latter
+initialization call if the underlying software runtime of \openshmem and MPI
+share the same internal communication resource.
+The program should always check the \VAR{provided} thread level returned
+at the corresponding initialization call or query the level of thread support
+after initialization to portably ensure thread support in each communication
+environment.
+
+Both \openshmem and MPI define similar thread levels, namely, \VAR{THREAD\_SINGLE},
+\VAR{THREAD\_FUNNELED}, \VAR{THREAD\_SERIALIZED}, and \VAR{THREAD\_MULTIPLE}.
+When requesting threading support in a hybrid program, however,
+the following additional rules are applied if the implementations of \openshmem
+and MPI share the same internal communication resource.
+Users are strongly advised to always follow these rules to ensure program
+portability.
+
+\begin{itemize}
+    \item The \VAR{THREAD\_SINGLE} thread level requires a single-threaded program.
+    Hence, users should not request \VAR{THREAD\_SINGLE} at the initialization
+    call of either \openshmem or MPI but request a different thread level at the
+    initialization call of the other model in the same program.
+
+    \item The \VAR{THREAD\_FUNNELED} thread level allows only the main thread to
+    make communication calls. A hybrid program using the \VAR{THREAD\_FUNNELED}
+    thread level in both \openshmem and MPI should ensure that the same main thread
+    is used in both communication environments.
+
+    \item The \VAR{THREAD\_SERIALIZED} thread level requires the program to ensure
+    that communication calls are not made concurrently by multiple threads. If a
+    hybrid program uses \VAR{THREAD\_SERIALIZED} in one communication environment
+    and \VAR{THREAD\_SERIALIZED} or \VAR{THREAD\_FUNNELED} in the other one, it
+    should also guarantee that the \openshmem and MPI calls are not made concurrently
+    from two distinct threads.
+\end{itemize}
+
+\subsection{Mapping Process Identification Numbers}
+\label{subsec:interoperability:id}
+
+Similar to the PE identifier in \openshmem, MPI defines rank as the
+identification number of a process in a communicator. Both the \openshmem PE
+and the MPI rank are unique integers assigned from zero to one less than the total
+number of processes. In a hybrid program, the \openshmem
+PE and the MPI rank in \VAR{MPI\_COMM\_WORLD} of a process can be equal.
+This feature, however, may be provided by only some of the \openshmem and MPI
+implementations (e.g., if both environments share the same underlying process
+manager) and is not portably guaranteed. A portable program should always
+use the standard functions in each model, namely, \FUNC{shmem\_my\_pe} in \openshmem
```

Comment:
```suggestion
use the standard functions in each model, namely, \FUNC{shmem\_team\_my\_pe} in \openshmem
```

Reply:
Also replaced the old API with teams-API in the example code.

Reply:
Also replaced the old API with teams-API in the example code.